clang/lib/AST/ExprConstant.cpp

```diff
@@ -1155,7 +1156,7 @@ namespace {
         if (CTSD->isInStdNamespace() && ClassII &&
             ClassII->isStr("allocator") && TAL.size() >= 1 &&
             TAL[0].getKind() == TemplateArgument::Type)
-          return {Call->Index, TAL[0].getAsType()};
+          return {Call->Index, TAL[0].getAsType(), Call->CallExpr};
       }
 
       return {};
```

Comment:
So `Call` will be `nullptr` here and following some of the code from `createHeapAlloc` I am not sure that is safe. Maybe the combination can't happen but it is not obvious to me.

Reply:
All callers of `getStdAllocatorCaller()` check that the returned value converts to `true`, and `Call` can't be`nullptr` if that's the case.